src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java

```diff
@@ -0,0 +1,84 @@
+package io.cdap.plugin.gcp.common;
+
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.ServiceOptions;
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.plugin.PluginConfig;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Contains config properties common to all GCP plugins, like project id and service account key.
+ */
+public class GCPConfig extends PluginConfig {
+  public static final String NAME_PROJECT = "project";
+  public static final String NAME_SERVICE_ACCOUNT_FILE_PATH = "serviceFilePath";
+  public static final String AUTO_DETECT = "auto-detect";
+
+  @Name(NAME_PROJECT)
+  @Description("Google Cloud Project ID, which uniquely identifies a project. "
+    + "It can be found on the Dashboard in the Google Cloud Platform Console.")
+  @Macro
+  @Nullable
+  protected String project;
+
+  @Name(NAME_SERVICE_ACCOUNT_FILE_PATH)
+  @Description("Path on the local file system of the service account key used "
+    + "for authorization. Can be set to 'auto-detect' when running on a Dataproc cluster. "
+    + "When running on other clusters, the file must be present on every node in the cluster.")
+  @Macro
+  @Nullable
+  protected String serviceFilePath;
+
```

Comment:
All public methods should have documentation

Reply:
done